examples/simulation/simulate_wire_chamber.cpp

```diff
@@ -0,0 +1,151 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/edm/track_parameters.hpp"
+#include "traccc/io/utils.hpp"
+#include "traccc/options/common_options.hpp"
+#include "traccc/options/handle_argument_errors.hpp"
+#include "traccc/options/options.hpp"
+#include "traccc/options/particle_gen_options.hpp"
+#include "traccc/options/propagation_options.hpp"
+#include "traccc/simulation/measurement_smearer.hpp"
+#include "traccc/simulation/simulator.hpp"
+#include "traccc/simulation/smearing_writer.hpp"
+
+// detray include(s).
+#include "detray/detectors/bfield.hpp"
+#include "detray/detectors/create_wire_chamber.hpp"
+#include "detray/io/common/detector_writer.hpp"
+#include "detray/simulation/event_generator/track_generators.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// Boost include(s).
+#include <boost/filesystem.hpp>
+
+using namespace traccc;
+namespace po = boost::program_options;
+
+int simulate(std::string output_directory, unsigned int events,
+             const traccc::particle_gen_options<scalar>& pg_opts,
+             const traccc::propagation_options<scalar>& propagation_opts) {
+
+    // Use deterministic random number generator for testing
+    using uniform_gen_t =
+        detray::random_numbers<scalar, std::uniform_real_distribution<scalar>,
+                               std::seed_seq>;
+
+    // Memory resource
+    vecmem::host_memory_resource host_mr;
+
+    /*****************************
+     * Build a wire chamber
+     *****************************/
+
+    // Detector type
+    using detector_type = detray::detector<detray::default_metadata>;
+
+    // B field value and its type
+    // @TODO: Set B field as argument
+    using b_field_t = covfie::field<detray::bfield::const_bknd_t>;
+    const vector3 B{0, 0, 2 * detray::unit<scalar>::T};
+    auto field = detray::bfield::create_const_field(B);
+
+    // Set Configuration
+    detray::wire_chamber_config wire_chamber_cfg{};
+    wire_chamber_cfg.n_layers(20u);
+
+    // Create the toy geometry
+    const auto [det, name_map] =
```

Comment:
Can this example be made universal by reading in a detector from file names that are passed as arguments to the main function?

Reply:
As I mentioned above, I would not like to maintain the test detector files in the repository

Reply:
Ok, I understand.